contracts/AelinPool.sol

```diff
@@ -70,6 +70,10 @@ contract AelinPool is AelinERC20, MinimalProxyFactory, IAelinPool {
         address _aelinDealLogicAddress,
         address _aelinRewardsAddress
     ) external initOnce {
+        require(_poolData.purchaseToken != address(0), "cant pass null purchaseToken address");
+        require(_sponsor != address(0), "cant pass null sponsor address");
+        require(_aelinDealLogicAddress != address(0), "cant pass null aelinDealLogicAddress");
+        require(_aelinRewardsAddress != address(0), "cant pass null aelinRewardsAddress");
```

Comment:
this makes sense to me. We dont care if anyone else calls initialize function because there should be an initOnce modifier and the contract that calls the initialize function is the one doing it from the contract that created it using the minimal proxy pattern